src/OIDCServiceProvider.php

```diff
@@ -53,8 +54,17 @@ final public function boot(): void
 
     private function getOIDCClient(): Client
     {
+        $this->autodiscovery();
+
         $config = collect(config('oidc'))
             ->put('redirect_uri', route('oidc.callback'));
         return new Client($config->all());
     }
+
+    private function autodiscovery(): void {
+        if (!is_null($providerURL = config('oidc.provider_url'))) {
+            $config = Http::get($providerURL)->json();
```

Comment:
This isn't optimal. You should check if the request were successfull and handle the error: https://laravel.com/docs/8.x/http-client#error-handling